src/python/pants/engine/build_files.py

```diff
@@ -44,8 +44,10 @@ def _key_func(entry):
   The AddressFamily may be empty, but it will not be None.
   """
   path_globs = PathGlobs(
-    include=(os.path.join(directory.path, p) for p in address_mapper.build_patterns),
-    exclude=address_mapper.build_ignore_patterns,
+    globs=(
+      *(os.path.join(directory.path, p) for p in address_mapper.build_patterns),
+      *(f"!{p}" for p in address_mapper.build_ignore_patterns),
+    )
```

Comment:
I view this particular change as a regression. But, presumably, these types of call sites will be limited and overall this change is a good one.

Reply:
Do you mean a performance regression? Is there reason to be concerned about that here?